### DISCUSSION TITLE
Basic question: How can I log Modbus messages from Sensorbox to SmartEVSE?

### DISCUSSION THREAD

**Discussion by negronoir** [serkri/SmartEVSE-3#222] [Q&A]
I run into a problem from time to time and I'm not able to find a solution.

I've a standard installation:

SmartEVSE v3 firm Serkri v1.7.3
Smart mode
1 phase
SensorBox plus
No battery, solar, nothing... Only grid

Usually works fine and power to my car is regulated dinamically due to Sensorbox data

But sometimes, for no apparent reason, SmartEVSE doesn't regulate and gives continuosly 32A to my car even though there is consumption at home. So circuit breaker is overcharged and some minutes later switches off

In the pic attached can be checked:

![20231017](https://github.com/serkri/SmartEVSE-3/assets/53239888/a58cd83c-77e3-4faa-8d9c-0d34ef122752)

-Total grid power available is 10kW

-Blue line: Tesla Car power from SmartEVSE.
Is about 2.5kW.
At 05:53 stops due to other consumptions not reflected in the grpah
At 06:01 restarts but no dynamic regulation (?) Goes up to 7.4kW (max set both car and SmartEVSE)
At 06:05 circuit breaker swithes off

-Purple line: Total grid usage
Remains just below 10kW all night but, when car restarts its charging at 6:01, goes ober 15kW for 4 minutes so circuit breaker switches off

I'm thinking about some bug in 1.7.3 firm but I can't guarantee. My only clue: Never had this problem with previous versions 

Any help?

Thanks!!

Toni

**Comment by dingo35** [comment]
Your headline and your post seem to be two different subjects.

But to answer the question:
Flash the debug version of the (most recent 1.8.0 ) firmware, telnet to your smartevse, press v, press enter (for verbose) and redirect your output to some log file.

**Reply by negronoir** [reply]
Thanks!

In fact I want to log modbus messages trying to find the problem :-)

**Comment by dingo35** [comment]
Also check if SmartEVSE is correctly detecting that you are charging on a single phase.

**Reply by negronoir** [reply]
Sure! It has been working for a year and a half and no changes beyond the firmware
And It's hard to find the problem because it happens only about twice a month

Thanks!

**Reply by dingo35** [reply]
[no_phase_detection_if_no_evmeter.zip](https://github.com/serkri/SmartEVSE-3/files/13685211/no_phase_detection_if_no_evmeter.zip)

You might try this version to rule out a phase detection problem, it disables phase detection if no EV meter is connected (as is the case in your config).

**Reply by negronoir** [reply]
Thanks! I have now 1.8.0 debug version dumping a verbose output to file. So I'll try this new version.

I'll write when the problem happens again. May be some days or some weeks...

**Reply by negronoir** [reply]
@dingo35  It seems to work but I'd like to test your change together with some minor changes not related to charging done by myself (e.g. my APHostname is a custom one)

Could you please share the change in the source included in "no_phase_detection_if_no_evmeter" version? to allow me rebuild it?

I'll continue to test tour change some more days before closing issue.

Thanks!

Toni

**Comment by dingo35** [comment]
It is in 8a1638e74b; if you compile current master branch it will be included.

**Reply by negronoir** [reply]
Merry Christmas!

I'll close the discussion. The issue seems to be fixed in your change.

Thanks

Toni
